client/components/search-themes/use-outside-click-callback.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useEffect } from 'react';
+import * as React from 'react';
+
+/**
+ * Hook that executes `callback` is the 'Escape' key is pressed
+ * or if the user clicks outside of `ref`.
+ *
+ * @param ref        Ref to an HTML element
+ * @param callback   Function to be executed
+ * @param useCapture useCapture flag passed to the add/removeEventListener
+ */
+export default function useOutsideClickCallback(
```

Comment:
I think we can put it under the `lib/use-outside-click-callback` and share it with `jetpack/profile-dropdown` and her

Reply:
I'll create a follow-up issue to have the `jetpack/profile-dropdown` component to import from this file as well (as I'm not sure if it'd trigger code review from other teams, or other deployment processes).

Reply:
I found another choice to use the `__experimentalUseFocusOutside` hook from Gutenberg

```js
import { __experimentalUseFocusOutside as useFocusOutside } from '@wordpress/compose';
```

Reply:
Changing to `useFocusOutside` seems to work nicely 👍